sync_performance_scaling_guide.adoc

```diff
@@ -1,66 +1,119 @@
+You can then send a HTTP GET request to the `/mbaas/sync/status` endpoint to view the summary of those metrics data. 
 
-  actual sync frequency = sync processor execution time + the sync frequency
+The following information is available from this endpoint:
 
+* CPU & Memory Usage of all the workers
+* The time taken to process various jobs
+* The number of the remaining jobs in various job queues
+* The time taken for various API calls
+* The time taken forvarious MongoDB operations
 
-This will help you to calculate the number of requests the system will make to the Dataset Backend.
+For each of those metrics, you will be able to see the total number of samples, the current, maximum, minimum and average values. 
 
-To know how long each sync processor execution takes, you can query the sync stats endpoint to see the average `Job Process Time` it takes for the `sync_worker` to complete.
+By default, it will collect the last 1000 samples for each metric, but you can control that using the `statsRecordsToKeep` configuration option.
 
-=== How much load can the Dataset Backend service handle?
+This endpoint is easy to use and will provide enough information for you to understand the current performance of the sync server. 
+However, if you want to visualize and understand the performance over a period of time, you should consider the next option.
 
-As mentioned above, everytime when the sync processor runs, it will perform a list operation on the Dataset Backend.
-When you setup the sync frequency, you should be able to estimate how many requests it will generate on your backend, and make sure the backend can handle the load.
+=== 2. Visualize the metrics data with InfluxDB and Grafana
 
-For example, if you set the sync frequency of a dataset to be 100ms, and each sync processor execution is taking 100ms to run, that means the server will generate about 5 req/sec to your backend.
-However, if you have another dataset with the same frequency that uses the same backend, it means there will be about 10 req/sec to the backend.
-The rate will be fairly consistent, so you can perform load tests against the backend to determine if your backend can handle that much load.
+If you want to visualize the current and historical metrics data, you can instruct the sync server to send the metrics data to InfluxDB, and see the graphs in Grafana.
 
-However, its worth mentioning that this value will not grow when you scale the app.
-For example, if you have multiple workers in your server, the sync processor executions are distributed among the workers rather than duplicated among them.
-This is by design to protect the backend when the app is under heavy load.
+There are plenty of tutorials online to help setup InfluxDB and Grafana. For example:
 
-=== How much extra load it will cause to the server itself?
+* http://www.netinstructions.com/installing-influxdb-and-grafana-on-an-ec2-instance/[How to setup InfluxDB and Grafana on AWS EC2]
```

Comment:
Might be better to just point to the 1 upstream example for setting up influx & grafana via docker/openshift.